api/client/trust.go

```diff
@@ -410,24 +428,15 @@ func (cli *DockerCli) trustedPush(repoInfo *registry.RepositoryInfo, tag string,
 		return err
 	}
 
-	for _, target := range targets {
-		h, err := hex.DecodeString(target.digest.Hex())
-		if err != nil {
-			return err
-		}
-		t := &client.Target{
-			Name: target.reference.String(),
-			Hashes: data.Hashes{
-				string(target.digest.Algorithm()): h,
-			},
-			Length: int64(target.size),
-		}
-		if err := repo.AddTarget(t, releasesRole); err != nil {
-			return err
-		}
+	if err := repo.AddTarget(target, releasesRole); err != nil {
+		return err
 	}
 
 	err = repo.Publish()
+	if err == nil {
```

Comment:
```
if err := repo.Publish(); err == nil {
    ...
```
Would also be good to add a clearer failure message, I've noticed we're not very good about messaging failure either, the lines just below this that check the error type end up printing a pretty terse message that isn't always clear.
```
} else if _, ok := err.(client.ErrRepoNotInitialized); !ok {
    fmt.Fprintf(cli.out, "Failed to sign %q:%s - %s\n", repoInfo.FullName(), tag, err.Error())
    return notaryError(repoInfo.FullName(), err)
}
```